Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1

```diff
@@ -0,0 +1,357 @@
+        $controls = $controlForRemediation.FailedControlSet
+        $resourceDetails = $controls | Where-Object { $controlIds -eq $_.ControlId};
+
+        if(($resourceDetails | Measure-Object).Count -eq 0 -or ($resourceDetails.ResourceDetails | Measure-Object).Count -eq 0)
+        {
+            Write-Host "No cloud service(s) found in input json file for remediation." -ForegroundColor $([Constants]::MessageType.Error)
+            break
+        }
+        $resourceDetails.ResourceDetails | ForEach-Object { 
+            try
+            {
+                $resources += Get-AzResource -ResourceId $_.ResourceId
+            }
+            catch
+            {
+                Write-Host "Valid resource group(s) or resource name(s) not found in input json file. ErrorMessage [$($_)]" -ForegroundColor $([Constants]::MessageType.Error)
+                break
+            }
+        }
+    }
+
+    $totalCloudService = ($resources | Measure-Object).Count
+    if($totalCloudService -eq 0)
+    {
+        Write-Host "No cloud service(s) found." -ForegroundColor $([Constants]::MessageType.Error);
+        Write-Host $([Constants]::DoubleDashLine)
+        break;
+    }
+
+    Write-Host "Total cloud service(s): [$($totalCloudService)]"
+    $folderPath = [Environment]::GetFolderPath("MyDocuments") 
+    if (Test-Path -Path $folderPath)
+    {
+        $folderPath += "\AzTS\Remediation\Subscriptions\$($subscriptionid.replace("-","_"))\$((Get-Date).ToString('yyyyMMdd_hhmm'))\DisableRemoteDesktopAccess"
+        New-Item -ItemType Directory -Path $folderPath | Out-Null
+    }
+
+    Write-Host "Checking config of cloud service(s) for remediation: [$(($resources | Measure-Object).Count)]"
+    
+    try {
+        $cloudServiceWithEnabledRDPAccess = @()
+        $cloudServiceWithDisabledRDPAccess = @()
+        $skippedCloudServiceFromRemediation = @()
+
+        # Checking cloud service(s) with enabled RDP acccess
+        $resources |  ForEach-Object {
+            $resource = $_
+            $rdpExtensions = $()
+            $cloudServiceExtension = Get-AzureServiceRemoteDesktopExtension -ServiceName $_.Name -ErrorAction SilentlyContinue
+            
+            $rdpExtensions += $cloudServiceExtension | Where-Object { ($null -ne $_) -and ($_.Extension -like "*RDP*") }
+            
+            if(($rdpExtensions | Measure-Object).Count -gt 0)
+            {
+                $item =  New-Object psobject -Property @{  
+                    CloudServiceName = $_.Name                
+                    ResourceGroupName = $_.ResourceGroupName
+                    RDPExtensionDetails = $rdpExtensions
+                }
+                    
+                $cloudServiceWithEnabledRDPAccess += $item        
+            }
+            else {
+                $cloudServiceWithDisabledRDPAccess += $_ | Select-Object @{Expression={($resource.ResourceGroupName)};Label="ResourceGroupName"},@{Expression={$resource.Name};Label="CloudServiceName"}
```

Comment:
There can be two cases:
- Extension is not present.
- Extension is present, but, disabled.
Do we consider both of them for the "disabled" use case?

Reply:
Covered this  scenario.